sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/UnionEstimation.scala

```diff
@@ -111,6 +111,44 @@ object UnionEstimation {
       AttributeMap.empty[ColumnStat]
     }
 
```

Comment:
The `estimate` method is a bit too long, so how about splitting it into smaller functions?

Reply:
Done. I have splitted it to smaller methods